align/constGap_lowMem3.go

```diff
@@ -0,0 +1,225 @@
+		route, routeIdx_current, i_inChecker_min, j_inChecker_min = WriteCigar(trace, i_inChecker_max, j_inChecker_max, route, routeIdx_current, i_inChecker_min, j_inChecker_min)
+
+		//Use Step 3's i_inChecker_min and j_inChecker_min to find the next checkerboard to go to and where to start in that checkerboard (update k1, k2)
+		if i_inChecker_min < 0 && j_inChecker_min < 0 {
+			k1, k2 = k1-1, k2-1 //go to the next checkerboard with lower i, lower j
+		} else if i_inChecker_min < 0 { //but j_inChecker != 0
+			k1 -= 1 //go to the next checkerboard with lower i, same j
+		} else if j_inChecker_min < 0 { //but i_inChecker != 0
+			k2 -= 1 //go to the next checkerboard with lower j, same i
+		}
+
+	}
+
+	//Step 4: write the last cigar entry
+	//This step is necessary because the row i=0 and the column j=0 are always stored in trace_prep, and never filled in in any checkerboard, but the cigar ends by the route going into 1 box/entry in either the row i=0 or the column j=0
+	if i_inChecker_min != -1 && j_inChecker_min == -1 { //indicating that Step 3 ended when k1 and k2 reached the smallest combination, and reached j=0, so the last cigar is a "2", e.g. if last cigar entry is the i=1, j=0 square
+		route, routeIdx_current = LastCigar(route, routeIdx_current, 2)
+	} else if i_inChecker_min == -1 && j_inChecker_min != -1 { //indicating that Step 3 ended when k1 and k2 reached the smallest combination, and reached i=0, so the last cigar is a "1", e.g. if last cigar entry is the i=0, j=1 square
+		route, routeIdx_current = LastCigar(route, routeIdx_current, 1)
+	} //no more "else" because the only situation left is if Step 3 ended when k1 and k2 reached the smallest combination, and reached both i=0 and j=0, aka the i=0 j=0 square, and there is no sequence there, so no cigar
+
+	//Final processing (reverse route) and return outputs
+	reverseCigar(route)
+	return score_highest, route
+}
+
+//Step 1
+func HighestScore(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapPen int64, checkersize int) (int64, int, int, [][]int64, [][]int64) {
+
+	mRowCurrent := make([]int64, len(beta)+1)
+	mRowPrevious := make([]int64, len(beta)+1)
+	var mColumn int = len(alpha) + 1
+	trace_prep_i := make([][]int64, int(len(alpha)/checkersize)+1) //trace_prep_i saves all rows (i) that are needed to initialize checkerboard tracebacks
+	trace_prep_j := make([][]int64, int(len(beta)/checkersize)+1)  //trace_prep_j saves all columns (j) that are needed to initialize checkerboard tracebacks
+	for idx := 0; idx < len(trace_prep_i); idx++ {
+		trace_prep_i[idx] = make([]int64, len(beta)+1)
+	}
+	for idx := 0; idx < len(trace_prep_j); idx++ {
+		trace_prep_j[idx] = make([]int64, len(alpha)+1)
+	}
+	var i, j int
+
+	for i = 0; i < mColumn; i++ {
+
+		for j = range mRowCurrent {
+			if i == 0 && j == 0 {
+				mRowCurrent[j] = 0
+				trace_prep_j[j/checkersize][i] = mRowCurrent[j] //it is implied here that j%checkersize==0. It must be saved in trace_prep_j
+			} else if i == 0 {
```

Comment:
Is this where the column has grown but we are at the starting row still? (assuming base 0, so row 1 is 0)

I'm trying to think through how we could have only i ==0 or only j == 0.